provider/pkg/schema/gen.go

```diff
@@ -628,6 +644,42 @@ func (ctx *context) gatherResourceType() error {
 			requiredInputs.Add(sdkName)
 		}
 	}
+
+	// ** Autonaming **
+	autonameSpecs := map[string]AutoNamingSpec{}
+
+	lookForField := func(fieldName string) func(map[string]pschema.PropertySpec) {
+		return func(properties map[string]pschema.PropertySpec) {
+			sdkName := ToSdkName(fieldName)
+			if propSpec, has := inputProperties[sdkName]; has && propSpec.Type == "string" {
+				autonameSpec := AutoNamingSpec{}
+				spec, ok := ctx.resourceSpec.Properties[fieldName]
+				if ok {
+					autonameSpec.MinLength = spec.MinLength.Val
+					autonameSpec.MaxLength = spec.MaxLength.Val
+				}
+				autonameSpecs[sdkName] = autonameSpec
+			}
+		}
+	}
+
+	autoNamePropLookupFuncs := []func(map[string]pschema.PropertySpec){
+		lookForField("Name"),
+		lookForField(resourceTypeName+"Name"),
```

Comment:
Related to the question about `primaryIdentifiers` - aren't those good enough to detect name properties, rather than relying on heuristics here?

Do you have stats on how good both approaches are?

Reply:
Yeah I started with `primaryIdentifiers`. This is the reason why the `AutoNamingSpecs` was setup to be a map since there are multiple primary identifiers. However, it seems a bunch of the names are not actually seen to be primary identifiers and there are significantly more false positives with including primary identifiers. I also tried a few combinations of using `createOnlyProperties`, `required` fields which also performed poorly. With the current heuristic I didn't find any false positives for autonaming candidates (I think we want to err towards being conservative here than allow for false positives).